microcosm_pubsub/tests/chain/test_chain.py

```diff
@@ -0,0 +1,71 @@
+from hamcrest import (
+    assert_that,
+    calling,
+    equal_to,
+    is_,
+    raises,
+)
+from marshmallow import ValidationError
+
+from microcosm_pubsub.chain import Chain
+from microcosm_pubsub.chain.decorators import extracts
+
+
+class TestChain:
+
+    def test_chain_returns_value(self):
+        chain = Chain(
+            lambda: 100,
+            lambda: 200,
+        )
+        assert_that(
+            chain.resolve(),
+            is_(equal_to(200)),
+        )
+
+    def test_chain_functions_can_pass_variables(self):
+        chain = Chain(
+            lambda arg: arg,
+        )
+        assert_that(
+            chain.resolve(arg=200),
+            is_(equal_to(200)),
+        )
+
+    def test_chain_functions_has_access_to_context(self):
+        chain = Chain(
+            lambda context: context["arg"],
```

Comment:
It might be nice if the context implemented `getattr` so we could use `context.arg`